ddl/index.go

```diff
@@ -312,7 +312,16 @@ func (w *worker) onCreateIndex(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int
 			return ver, errors.Trace(err)
 		}
 
-		err = w.runReorgJob(t, reorgInfo, d.lease, func() error {
+		err = w.runReorgJob(t, reorgInfo, d.lease, func() (addIndexErr error) {
+			defer func() {
+				r := recover()
```

Comment:
use util.WithRecover instead.

Reply:
No, I need return cancelled error here.

Reply:
you can change `util.WithReocover` to return value from `exec` func

Reply:
Ok, go WithReocover can not return values